packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3235 @@
+  final double elevation;
+
+  /// The minimum height to give the menu bar.
+  final double height;
+
+  /// The background color of the menu app bar.
+  final Color color;
+
+  /// The padding around the outside of the menu bar contents.
+  final EdgeInsets padding;
+
+  @override
+  final Size preferredSize;
+
+  /// The list of widgets to use as children of this menu bar.
+  ///
+  /// These are the top level [MenuBarMenu]s.
+  final List<MenuItem> children;
+
+  @override
+  Widget build(BuildContext context) {
+    final _MenuBarController controller = MenuBarController.of(context) as _MenuBarController;
+
+    int index = 0;
+    final Widget appBar = Material(
+      elevation: elevation,
+      color: color,
+      child: Row(
+        children: <Widget>[
+          ...children.map<Widget>((MenuItem child) {
+            final Widget result = _MenuNodeWrapper(
+              serial: controller.menuSerial,
+              menu: controller.root.children[index],
+              child: child as Widget,
+            );
+            index += 1;
+            return result;
+          }).toList(),
+          const Spacer(),
+        ],
+      ),
+    );
+
+    return ConstrainedBox(
+      constraints: BoxConstraints(minHeight: height),
+      child: appBar,
+    );
+  }
+}
+
+/// A label widget that is used as the default label for a [MenuBarItem] or
+/// [MenuBarMenu].
+///
+/// It not only shows the [MenuBarMenu.label] or [MenuBarItem.label], but if
+/// there is a shortcut associated with the [MenuBarItem], it will display a
+/// mnemonic for the shortcut. For [MenuBarMenu]s, it will display a visual
+/// indicator that there is a submenu.
+class _MenuBarItemLabel extends StatelessWidget {
+  /// Creates a const [_MenuBarItemLabel].
+  ///
+  /// The [menuBarItem] argument is required.
+  const _MenuBarItemLabel({
+    this.leadingIcon,
+    required this.label,
+    this.trailingIcon,
+    this.shortcut,
+    required this.hasSubmenu,
+  });
+
+  /// The optional icon that comes before the [label].
+  final Widget? leadingIcon;
+
+  /// The required label widget.
+  final Widget label;
+
+  /// The optional icon that comes after the [label].
+  final Widget? trailingIcon;
+
+  /// The shortcut for this label, so that it can generate a string describing
+  /// the shortcut.
+  final MenuSerializableShortcut? shortcut;
+
+  /// Whether or not this menu has a submenu.
+  final bool hasSubmenu;
+
+  @override
+  Widget build(BuildContext context) {
+    final _MenuBarController controller = MenuBarController.of(context) as _MenuBarController;
+    final bool isTopLevelItem = _MenuNodeWrapper.of(context).parent == controller.root;
+    final VisualDensity density = Theme.of(context).visualDensity;
+    final double horizontalPadding = math.max(
+      _kLabelItemMinSpacing,
+      _kLabelItemDefaultSpacing + density.horizontal * 2,
+    );
+    return AnimatedBuilder(
+      animation: controller,
```

Comment:
Surprised we need to rebuild when controller notifies its listeners since nothing below reads off controller.

Reply:
Ah, is it because of isTopLevelItem? Maybe add a comment

Reply:
I had to think about it a bit, but actually, I think you're right.  I don't think topLevelItem could change, since it would involve the _MenuNode changing. I used to have other things that needed updating when the current open menu changed, but the code got simpler, so it doesn't need that anymore.